frontend/libs/studio-components/src/components/StudioGridSelector/StudioGridSelector.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import classes from './StudioGridSelector.module.css';
+import cn from 'classnames';
+
+type StudioGridSelectorProps = {
+  disabled?: boolean;
+  sliderValue?: number;
+  handleSliderChange: (newValue: number) => void;
+};
+
+/**
+ * @component
+ *    Component to select a value within the range 1 to 12. Adapted to select value for grid-property on layout components
+ */
+export const StudioGridSelector = ({
+  disabled = false,
+  sliderValue = 12,
+  handleSliderChange,
+}: StudioGridSelectorProps) => {
+  const [value, setValue] = useState<number>(sliderValue);
+  const gridValues = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+  useEffect(() => {
+    setValue(sliderValue ?? 12);
```

Comment:
Should we skip the fallback to 12 in this context, given that we have already established 12 as the default value during destructuring?

```suggestion
    setValue(sliderValue);
```